api/chaos/serverexperimentation/v1/serverexperimentation.proto

```diff
@@ -56,3 +56,128 @@ enum FaultInjectionCluster {
+  FaultAbortStatus fault_abort_status = 3 [ (validate.rules).message.required = true ];
+}
+
+// The definition of a latency fault.
+message LatencyFault {
+  // The targeting of the fault describing where the faults should be applied.
+  FaultTargeting fault_targeting = 1 [ (validate.rules).message.required = true ];
+  // The percentage of requests the fault should be applied to.
+  FaultPercentage fault_parcentage = 2 [ (validate.rules).message.required = true ];
+  // The latency duration to apply.
+  FaultLatencyDuration fault_latency_duration = 3 [ (validate.rules).message.required = true ];
+}
+
+// The fault targeting that describes a part of the system that's responsible for enforcing a fault.
+message FaultTargeting {
+  // A fault enforcer.
+  oneof enforcer {
+    option (validate.required) = true;
+
+    // Enforce a fault on an upstream.
+    UpstreamEnforcing upstream = 1;
+    // Enfore a fault on a downstream.
+    DownstreamEnforcing downstream = 2;
+  }
+}
+
+/// Enforce faults on an upstream.
+message UpstreamEnforcing {
+  // An upstream responsible for enforcing a fault.
+  oneof upstream_type {
+    option (validate.required) = true;
+
+    // A single upstream cluster.
+    SingleCluster upstream_cluster = 1;
+    // A partial single upstream cluster.
+    PartialSingleCluster upstream_partial_single_cluster = 2;
+  }
+
+  // A downstream responsible for enforcing a fault.
+  oneof downstream_type {
+    option (validate.required) = true;
+
+    // A single downstream cluster.
+    SingleCluster downstream_cluster = 3;
+  }
+}
+
+/// Enforce faults on a downstream.
+message DownstreamEnforcing {
+  // An upstream responsible for enforcing a fault.
+  oneof upstream_type {
+    option (validate.required) = true;
+
+    // A single upstream cluster.
+    SingleCluster upstream_cluster = 1;
+  }
+
+  // A downstream responsible for enforcing a fault.
+  oneof downstream_type {
+    option (validate.required) = true;
+
+    // A single downstream cluster.
+    SingleCluster downstream_cluster = 2;
+  }
+}
+
+// A single cluster.
+message SingleCluster {
+  // The name of a cluster.
+  string name = 1 [ (validate.rules).string = {min_bytes : 1} ];
+}
+
+// A partial single cluster - the part of a single cluster.
+message PartialSingleCluster {
+  // The name of a cluster.
+  string name = 1 [ (validate.rules).string = {min_bytes : 1} ];
+  // The percentage controlling what part of a cluster should be included.
+  ClusterPercentage cluster_percentage = 2 [ (validate.rules).message.required = true ];
+}
+
+// The percentage controlling what portion of a given cluster should be applying faults.
+message ClusterPercentage {
+  // The numerator of a percentage with a fixed denumerator equal to 100
+  // (i.e. percentage equal to 50 results in 50/100 = 50%)
+  uint32 percentage = 1 [ (validate.rules).uint32 = {gt : 0, lte : 100} ];
+}
+
+// The fault percentage controlling how often a given fault should be applied.
```

Comment:
"Often" to me implies a temporal aspect, I would phrase this is `what percentage of requests should have the fault applied`